src/EditorFeatures/Test/Completion/AbstractCompletionProviderTests.cs

```diff
@@ -53,27 +53,43 @@ public override void Dispose()
             return !service.GetMemberBodySpanForSpeculativeBinding(node).IsEmpty;
         }
 
-        internal static ICompletionService GetCompletionService(Document document)
+        internal static CompletionService GetCompletionService(Document document)
         {
-            return document.Project.LanguageServices.GetService<ICompletionService>();
+            return document.Project.LanguageServices.GetService<CompletionService>();
         }
 
-        internal static CompletionRules GetCompletionRules(Document document)
+        internal static CompletionHelper GetCompletionHelper(Document document)
         {
-            return GetCompletionService(document).GetCompletionRules();
+            var service = GetCompletionService(document);
+            return new CompletionHelper(service.GetRules(), document.Project.Language, service);
         }
 
-        internal static async Task<CompletionList> GetCompletionListAsync(CompletionListProvider provider, Document document, int position, CompletionTriggerInfo triggerInfo, OptionSet options = null)
+        internal static async Task<CompletionContext> GetCompletionListContextAsync(CompletionProvider provider, Document document, int position, CompletionTrigger triggerInfo, OptionSet options = null)
         {
             options = options ?? document.Project.Solution.Workspace.Options;
-            var context = new CompletionListContext(document, position, triggerInfo, options, CancellationToken.None);
+            var service = document.Project.LanguageServices.GetService<CompletionService>();
+            var text = await document.GetTextAsync();
+            var span = service.GetDefaultItemSpan(text, position);
```

Comment:
if this is called from an async context, why not make it async itself and pass along the document to it?  If it needs hte source text, it can ask for it.